akka-remote/src/main/scala/akka/remote/artery/TaskRunner.scala

```diff
@@ -91,14 +94,29 @@ private[akka] class TaskRunner(system: ExtendedActorSystem) extends Runnable {
   private[this] val tasks = new ArrayBag[Task]
 
   // TODO the backoff strategy should be measured and tuned
-  private val spinning = 2000000
-  private val yielding = 0
-  private val idleStrategy = new BackoffIdleStrategy(
-    spinning, yielding, TimeUnit.MICROSECONDS.toNanos(1), TimeUnit.MICROSECONDS.toNanos(100))
+  private val idleStrategy: IdleStrategy = {
+    if (idleCpuLevel == 1) {
+      val maxParkMicros = 400
+      new BackoffIdleStrategy(1, 1, TimeUnit.MICROSECONDS.toNanos(1), TimeUnit.MICROSECONDS.toNanos(maxParkMicros))
+    } else if (idleCpuLevel == 10)
+      new BusySpinIdleStrategy
+    else {
+      val spinning = 100000 * idleCpuLevel
+      val yielding = 2 * idleCpuLevel
+      val maxParkMicros = 40 * (11 - idleCpuLevel)
```

Comment:
probably can be more aggressive here if level is low (i.e. larger spread of the values)